build/phase.go

```diff
@@ -118,20 +120,17 @@ func (p *Phase) Run(context context.Context) error {
 		return errors.Wrapf(err, "failed to create '%s' container", p.name)
 	}
 	p.appOnce.Do(func() {
-		var mode int64 = -1
-		if runtime.GOOS == "windows" {
-			mode = 0777
+		var appReader io.ReadCloser
+		appReader, err = p.createAppReader()
+		if err != nil {
+			err = errors.Wrapf(err, "create tar archive from '%s'", p.appPath)
```

Comment:
We should return from this function if creating the tar reader fails instead of continuing to the copy

Reply:
Good catch!

Reply:
@ekcasey Resolved by adding err check around the copy. We can't return the error because we're in a closure, but the effect should be the same. Feel free to double-check us.

Reply:
we can't return an error but we can just `return` right?

Reply:
Yeah that's what we did

Reply:
Done.